Gifski/util.swift

```diff
@@ -2515,3 +2515,18 @@ extension NSObject {
 		return type(of: self).simpleClassName
 	}
 }
+
+extension AVPlayerItem {
+	/// The playable range of the item.
+	/// Can be `nil` when the `.duration` is not available, for example, when the asset has not yet been fully loaded or if it's a live stream.
+	var playbackRange: ClosedRange<Double>? {
+		guard duration.isNumeric else {
+			return nil
+		}
+
+		let startTime = reversePlaybackEndTime.isNumeric ? reversePlaybackEndTime.seconds : 0
```

Comment:
Instead of checking if the seconds are `nan`, we check that the CMTime is numeric and valid, which is a more correct way to handle it, and if not, we fall back to the full range.

Reply:
ah nice catch 👍